pkg/controller/cassandracluster/reconcile.go

```diff
@@ -169,113 +182,123 @@ func (rcc *ReconcileCassandraCluster) CheckNonAllowedChanged(cc *api.CassandraCl
 	return false
 }
 
-//CheckNonAllowedRemoveDC checks to see if the Operator accepts or refuses the CRD changes
-func CheckNonAllowedRemoveDC(rcc *ReconcileCassandraCluster, cc *api.CassandraCluster,
-	status *api.CassandraClusterStatus, oldCRD *api.CassandraCluster) (bool, string) {
-	//Check if we ask to remove DC/Rack and If we are allow to do it
-	if cc.GetDCRackSize() < oldCRD.GetDCRackSize() {
-		//We ask a decrease in the amount of DCRack
-		dcsize := cc.GetDCSize()
-		olddcsize := oldCRD.GetDCSize()
-		if dcsize < olddcsize-1 {
-			logrus.WithFields(logrus.Fields{"cluster": cc.Name}).Warningf(
-				"The Operator has refused the Topology changed. You can only remove 1 DC at a time, "+
-					"not only a Rack: %v restored to %v", cc.Spec.Topology, oldCRD.Spec.Topology)
-			cc.Spec.Topology = oldCRD.Spec.Topology
-			return true, api.ActionCorrectCRDConfig
+func generatePaths(s string) []string {
+	return strings.Split(s, ".")
+}
+
+func lookForFilter(path []string, filtersRule [][]string, filtersFound *map[string]bool) {
+	for _, filter := range filtersRule {
+		if 2*len(filter)+1 == len(path) {
+			currentPath := path[0]
+			for i := 2; i < len(path)-1; i += 2 {
+				currentPath += "." + path[i]
+			}
+			if currentPath == strings.Join(filter, ".") {
+				if _, ok := (*filtersFound)[currentPath]; !ok {
+					(*filtersFound)[currentPath] = true
+				}
+			}
 		}
+	}
+}
 
-		//If we ask to remove only a rack, then it is not authorized
-		if dcsize == olddcsize {
+// hasChange returns if there a the changes with the type provided and only for all paths
+// paths must be provided in ascending order like ('DC', 'DC.Rack') in order to work
```

Comment:
in the usage, you uses (.., 'DC.Rack', '-DC') so it's not always the ascending order ?